addons/mass_mailing/models/mailing.py

```diff
@@ -58,6 +58,16 @@ def _get_default_mail_server_id(self):
     email_from = fields.Char(string='Send From', required=True,
         default=lambda self: self.env.user.email_formatted)
     sent_date = fields.Datetime(string='Sent Date', copy=False)
+
+    def _default_schedule_type(self):
+        default_schedule_date = self.env.context.get('default_schedule_date')
+        if default_schedule_date:
+            return "send_on"
+        return "send_asap"
```

Comment:
Would be cleaner to implement this using a default_get.
You can check how it's done on the social_post model.